src/hooks/Circles/usePrivatePosts.ts

```diff
@@ -64,6 +64,40 @@ export const postToMessage = (
       avatar: post.author?.profile.picture,
     },
   };
+
+  const markdownImages = message.text.match(/\!\[.+\]\(.+\)/g);
+
+  if (!markdownImages) {
+    return message;
+  }
+
+  const messages: IMessage[] = [];
+  let text = message.text;
+
+  for (let i = 0; i < markdownImages.length; i++) {
+    const imageMarkdown = markdownImages[i];
+    const imageUrl = imageMarkdown.match(/\!\[.+\]\((.+)\)/)?.[1];
+    const parts = text.split(imageMarkdown);
+    text = parts[1].trim();
+
+    messages.push({
+      ...message,
+      _id: `${message._id}-${i}`,
+      text: parts[0].trim(),
+    });
+    messages.push({
+      ...message,
+      _id: `${message._id}-${i}-image`,
+      text: '',
+      image: imageUrl,
+    });
```

Comment:
I'm not familiar with the repo much less this feature, but this seems odd to me. You're adding the message twice and just changing whether it's image or text? Why not just

```suggestion
    messages.push({
      ...message,
      _id: `${message._id}-${i}`,
      text: parts[0].trim(),
      image: imageUrl,
    });
```

Reply:
Gifted chat renders the text below the image. For markdown images, we want them to appear in the order they were written so this splits the message up by image occurrences and interleaves them correctly.

Example: For the following markdown
```
Picture 1
![some-pic](https://some-url.com/1)
Picture 2
![some-pic-2](https://some-url.com/2)
Aren't these cool?
```

If we put the image on the message, gifted chat would display

<table>
<tr><td>&lt;some-pic-1&gt;<br>Picture 1
<tr><td>&lt;some-pic-2&gt;<br>Picture 2<br>Aren't these cool?
</table>

With my proposed solution we would see 

<table>
<tr><td>Picture 1
<tr><td>&lt;some-pic-1&gt;
<tr><td>Picture 2
<tr><td>&lt;some-pic-2&gt;
<tr><td>Aren't these cool?
</table>